x/mongo/driver/topology/polling_srv_records_test.go

```diff
@@ -371,6 +371,7 @@ func TestPollSRVRecordsMaxHosts(t *testing.T) {
 		compareHosts(t, actualHosts, expectedHosts)
 	})
 	t.Run("SRVMaxHosts is less than number of hosts", func(t *testing.T) {
+		t.Skipf("skipping due to GODRIVER-2222")
```

Comment:
Without the call to `rand.Seed`, the same subset of hosts is selected. Here is test output with logs:

```
go test -count=1 -v ./x/mongo/driver/topology -run TestPollSRVRecordsMaxHosts/SRVMaxHosts_is_less_than_number_of_hosts
=== RUN   TestPollSRVRecordsMaxHosts
=== RUN   TestPollSRVRecordsMaxHosts/SRVMaxHosts_is_less_than_number_of_hosts
Before shuffle
 -  localhost.test.build.10gen.cc:27017
 -  localhost.test.build.10gen.cc:27019
 -  localhost.test.build.10gen.cc:27020
After shuffle
 -  localhost.test.build.10gen.cc:27020
 -  localhost.test.build.10gen.cc:27017
Before shuffle
 -  localhost.test.build.10gen.cc:27017
 -  localhost.test.build.10gen.cc:27019
 -  localhost.test.build.10gen.cc:27020
After shuffle
 -  localhost.test.build.10gen.cc:27020
 -  localhost.test.build.10gen.cc:27017
--- PASS: TestPollSRVRecordsMaxHosts (0.08s)
    --- PASS: TestPollSRVRecordsMaxHosts/SRVMaxHosts_is_less_than_number_of_hosts (0.08s)
PASS
ok      go.mongodb.org/mongo-driver/x/mongo/driver/topology     0.713s
```

With the call to `rand.Seed`, if a subset is chosen that does not include `localhost:27017`, the test fails.